pype/hosts/maya/api/__init__.py

```diff
@@ -0,0 +1,226 @@
+                 "save/open/new callback installation..")
+        return
+
+    avalon.on("save", on_save)
+    avalon.on("open", on_open)
+    avalon.on("new", on_new)
+    avalon.before("save", on_before_save)
+
+    log.info("Overriding existing event 'taskChanged'")
+    override_event("taskChanged", on_task_changed)
+
+    log.info("Setting default family states for loader..")
+    avalon.data["familiesStateToggled"] = ["imagesequence"]
+
+
+def uninstall():
+    pyblish.deregister_plugin_path(PUBLISH_PATH)
+    avalon.deregister_plugin_path(avalon.Loader, LOAD_PATH)
+    avalon.deregister_plugin_path(avalon.Creator, CREATE_PATH)
+
+    menu.uninstall()
+
+
+def override_event(event, callback):
+    """
+    Override existing event callback
+    Args:
+        event (str): name of the event
+        callback (function): callback to be triggered
+
+    Returns:
+        None
+
+    """
+
+    ref = weakref.WeakSet()
+    ref.add(callback)
+
+    pipeline._registered_event_handlers[event] = ref
+
+
+def on_init(_):
+    avalon.logger.info("Running callback on init..")
+
+    def safe_deferred(fn):
+        """Execute deferred the function in a try-except"""
+
+        def _fn():
+            """safely call in deferred callback"""
+            try:
+                fn()
+            except Exception as exc:
+                print(exc)
+
+        try:
+            utils.executeDeferred(_fn)
+        except Exception as exc:
+            print(exc)
+
+    # Force load Alembic so referenced alembics
+    # work correctly on scene open
+    cmds.loadPlugin("AbcImport", quiet=True)
+    cmds.loadPlugin("AbcExport", quiet=True)
+
+    # Force load objExport plug-in (requested by artists)
+    cmds.loadPlugin("objExport", quiet=True)
+
+    from .customize import (
+        override_component_mask_commands,
+        override_toolbox_ui
+    )
+    safe_deferred(override_component_mask_commands)
+
+    launch_workfiles = os.environ.get("WORKFILES_STARTUP")
+
+    if launch_workfiles:
+        safe_deferred(launch_workfiles_app)
+
+    if not IS_HEADLESS:
+        safe_deferred(override_toolbox_ui)
+
+
+def launch_workfiles_app():
+    workfiles.show(os.environ["AVALON_WORKDIR"])
+
+
+def on_before_save(return_code):
+    """Run validation for scene's FPS prior to saving"""
+    return lib.validate_fps()
+
+
+def on_save():
+    """Automatically add IDs to new nodes
+
+    Any transform of a mesh, without an existing ID, is given one
+    automatically on file save.
+    """
+
+    avalon.logger.info("Running callback on save..")
+
+    # # Update current task for the current scene
+    # update_task_from_path(cmds.file(query=True, sceneName=True))
+
+    # Generate ids of the current context on nodes in the scene
+    nodes = lib.get_id_required_nodes(referenced_nodes=False)
+    for node, new_id in lib.generate_ids(nodes):
+        lib.set_id(node, new_id, overwrite=False)
+
+
+def on_open():
```

Comment:
Function must expect one argument so the callback crashes.
- was `def on_open(_):`

Reply:
thanks. I deleted them accidentally when moving it.